main.tf

```
@@ -14,15 +14,15 @@ locals {
   )}"
   atlantis_url_events = "${local.atlantis_url}/events"
 
-  # Include only one group of secrets - for github, gitlab or bitbucket
-  has_secrets = var.atlantis_gitlab_user_token != "" || var.atlantis_github_user_token != "" || var.atlantis_bitbucket_user_token != ""
+  # Include only one group of secrets - for github, github app,  gitlab or bitbucket
+  has_secrets = var.atlantis_gitlab_user_token != "" || var.atlantis_github_user_token != "" || var.atlantis_github_app_key != "" || var.atlantis_bitbucket_user_token != ""
```

I believe we can wrap all these in this (untested):
```suggestion
  has_secrets = try(coalesce(var.atlantis_gitlab_user_token, var.atlantis_github_user_token, var.atlantis_github_app_key, var.atlantis_bitbucket_user_token) != "", false)
```

---

or perhaps better yet:

```suggestion
  has_secrets = compact([var.atlantis_gitlab_user_token, var.atlantis_github_user_token, var.atlantis_github_app_key, var.atlantis_bitbucket_user_token]) > 0
```

---

I tested both suggestions, but only the first one went well. I made a commit with the proposed change, ran it locally and it worked fine with my test repo.